covenant/covenant.go

```diff
@@ -157,7 +157,30 @@ func (ce *CovenantEmulator) AddCovenantSignatures(btcDels []*types.Delegation) (
 			continue
 		}
 
-		// 4. decode staking tx and slashing tx from the delegation
+		// 4. check whether the:
```

Comment:
What about having a separate function to validate the btcDel with params

Reply:
so for now I went for the simplest possible implementation because there this ticket - https://github.com/babylonlabs-io/covenant-emulator/issues/3 to improve the whole thing.

My current thinking is that maybe, we can re-use functiond introduced in https://github.com/babylonlabs-io/babylon/pull/33 to do the validation here in the future, by for example:
- generalising ` ParseCreateDelegationMessage` to parse also delegation
- then re-use `ValidateParams` to do validation also at `covenant-emulator` level

Reply:
nice, understood!